compiler/src/main/java/com/squareup/anvil/compiler/IrUtils.kt

```diff
@@ -105,9 +108,23 @@ internal fun IrConstructorCall.parentScope(): FqName {
       element = this
     )
 
-  return FqName("${signature.packageFqName}.${signature.shortName}")
+  return signature.fqName
 }
 
+private val CommonSignature.fqName: FqName
+  // It's important that we use [declarationFqName] instead of [shortName] here to ensure that the
+  // [FqName] being constructed handles nested classes if it's later used for finding IR symbols.
+  //
+  // e.g. we might have a signature like
+  // 'com.squareup.anvil.test/ContributedSubcomponent.ParentScope' which gets broken up as
+  // packageFqName     -> 'com.squareup.anvil.test'
+  // declarationFqName -> 'ContributedSubcomponent.ParentScope'
+  // shortName         -> 'ParentScope'
+  //
+  // Using [shortName] would produce the FqName 'com.squareup.anvil.test.ParentScope' which
+  // prevents the IR lookup utils from finding the matching symbol.
```

Comment:
@vRallev do you think this comment is useful? It took a long time to track down the issue that I ran into related to this, although I'm not sure how likely it would be for someone to see this code and want to change it to use `shortName` later on 🤔

Reply:
I think it's useful! The APIs for the compiler are quite consistent and `shortName` always refers to THIS class name. But a comment like this certainly doesn't hurt!